### ISSUE TITLE
Squad Tag needs work A.S.A.P. (LOGO NOT APPEARING FIX HERE)

### ISSUE THREAD

**Issue by biowire20** [ins0/armasquads#60]
A large issue for many people here is that there picture or logo is not being displayed and you have the file in either jpg, png, or gif and the dimensions are either 16x16, 32x32, 128x128, and so on. Yet your logo does not appear the fix for this problem is that Arma squad xml's do not render your logo if your SQUAD TAG has a period in it  "." so to fix this if your clans tag is "W.S.C." you need your squad tag to be "WSC" and it will render your logo in-game if you have met the other to requirements I mentioned above. That is all and "ind0" if you read this, when some one is typing there squad tag and they put "." make it so that the website says they cant use a period in squad tag if possible.

**Comment by ins0** [comment]
thanks for your suggest @biowire20. yeah that is the thing why i mentioned to not enter characters other then ``a-z 0-9``. unfortunately i can't restrict this values from being entered as many users don't had a squad logo but would like to use squad tags like ``F.O.O`` or ``B.A.R``. so there is a usecase for both possibilities as arma render both equal (expects the squad logo with periods in tag).

i will try to make a hint if you enter squad tags with special characters that may your ingame logo will not appear.

**Comment by mwpowellhtx** [comment]
A period? Or any punctuation, let's say, i.e. `"... (SOC)"`? So remove the parens and try again perhaps? Tried that, still does not work. Would you mind clarifying, the image and tag names must agree?

**Comment by mwpowellhtx** [comment]
Pretty sure it is an issue in your image conversion. I can promise you I am saving a PNG and/or a JPG file as a 256x, however, the PAA, as downloaded from your site, is 255x ... suggested workarounds? upload as a proper PAA, perhaps? bypass any site based conversions?

**Comment by ins0** [comment]
@mwpowellhtx please provide a sample file png + file sha hash where u think the paa conversation is converting wrong.

**Comment by mwpowellhtx** [comment]
I don't know what you mean by SHA vis-a-vis the unit emblem, however, end-to-end, here is the source image, and result PAA. do you need to know the squads bits as well, preferably private not open forum such as this?

[b024c2d38c4c2cfe4006418bbeb8e2cf.zip](https://github.com/ins0/armasquads/files/12515940/b024c2d38c4c2cfe4006418bbeb8e2cf.zip), _the PAA file zipped_.

The source image:

![official_26th_meu_logo_256x_sepia](https://github.com/ins0/armasquads/assets/1824187/1e613dad-8c77-4d19-8bf4-2cbf3f2bd5af)

Which also happenstance converts to from just fine using the awesome [_web converter_](https://paa.gruppe-adler.de/). So, not sure how that is not connecting the dots in game in the three known scenarios, profile thumbnail, in game player tab, uni patch.

By comparison/contrast, the BIS A3 units also seems to be plagued with a couple of issues itself, getting 1/3 success thumbnail only.

HTH

**Comment by mwpowellhtx** [comment]
Wild thought, are there any ports I need to poke through my router situation in order for the assets to become visible?

```
{ "Cable modem", "Primary router", "Secondary router" }
```

Although I cannot imagine why this would be the case, but who knows, it is not uncommon. Hosting a TS server, for example.

**Comment by mwpowellhtx** [comment]
Which to be fair, using dashboard unit slot 1, in sequence, thumbnail, player tab, uni patch:

![image](https://github.com/ins0/armasquads/assets/1824187/7514f596-a8a6-46fe-ab2a-0e61586b57c0)
![image](https://github.com/ins0/armasquads/assets/1824187/73b4e46b-d676-44af-b5ee-6781a2237815)
![image](https://github.com/ins0/armasquads/assets/1824187/9f82cd69-c5ca-4bdd-93dc-184bf937de5d)

Then with squads custom configured:

![image](https://github.com/ins0/armasquads/assets/1824187/234bfca5-507e-430e-a9ed-7233a47aebd3)
![image](https://github.com/ins0/armasquads/assets/1824187/fe8f640a-313f-4969-a2ce-65f1caf6b620)

To be fair, profile thumbs seemed to be working just the other day, now are not working. So maybe there is a legit issue with BIS arma units servers or something, who knows. Still, why dashboard units are working, sans the profile thumbs, and not squads, is a mystery.

Logo renditions notwithstanding, I can tell you that the tags are at least appending correctly in both cases.

**Comment by ins0** [comment]
@mwpowellhtx i'm currently upgrading the server - give me 15 minutes and i can ping you with an update and than you could test again - thanks!

**Comment by mwpowellhtx** [comment]
Sure no worries any way I can help.

**Comment by ins0** [comment]
@mwpowellhtx give it a try pls. It's required that you delete or create a new squad logo that the changes takes effect.

**Comment by mwpowellhtx** [comment]
@ins0 thanks for the turnaround, understood. I guess question while we're here, any way I can change my account email address? There is an obsolete one on record, really want for that to be updated. However, attempts to create a new account were blocked for whatever reason, IIRC, names or ID handles or whatever were also blocking that.

**Comment by mwpowellhtx** [comment]
> @mwpowellhtx give it a try pls. It's required that you delete or create a new squad logo that the changes takes effect.

Perhaps I misunderstood. Drop the old logo on file and upload a _"new"_ one (if the same one)?

**Comment by ins0** [comment]
Yes delete the logo and then edit the squad and upload a new one. Sure thing pls email me you old and new email to "ins0@racecore.de"

**Comment by mwpowellhtx** [comment]
Still no joy this side. Profile thumb bit of a question. No joy player tab, nor for uni patch.

**Comment by ins0** [comment]
Could you please guide me to everything that exists in the squad logo
cosmos - i'm out playing arma and from by point you can only set a custom
url in your online arma profile. so i'm kind of confused what now works and
what doesn't since i'm not familiar with new services etc by bohemia.

Am Di., 5. Sept. 2023 um 01:11 Uhr schrieb Michael W Powell <
***@***.***>:

**Comment by mwpowellhtx** [comment]
@ins0 So to recap. What we discovered, there is a transient path landing in the A3 `AppData\Local` folder. Should probably be `.\squads\>>tag<<\file.paa`, but is actually something more like `.\Tmp2302\squads\>>tag<<\file.paa`, something like that, I may have the digits transposed. But at any rate, fix that, whack the temp directory, and my suspicion is that the problem sorts itself. Otherwise, unit tags are appending just fine it seems, but for the file directory snafu. Glad I could be helpful.

**Comment by mwpowellhtx** [comment]
@ins0 How goes it? Educated guess on my part, [simple search of keyword `"Tmp"`](https://github.com/search?q=repo%3Ains0%2Farmasquads+Tmp&type=code), I generate this result. Looks like `SquadsController` is a bit suspect with some sort of temp file path, perhaps that is finding its way somehow into the ARMA request response?

And that it is always, consistently `"Tmp2302"`, that tells me perhaps it is something [configured on your service and/or database portal backend](https://github.com/ins0/armasquads/blob/a8879d68f7f48abb67e8a4ffd6342676560d6c01/src/module/Frontend/Api/src/Frontend/Api/v1/Controller/SquadsController.php#L98).

Good luck, good hunting!

**Comment by mwpowellhtx** [comment]
**Update**: IDK what the issue is exactly, when I land in a proper dedi server scenario, my squads XML shows hundred percent as expected, and no matter what from that point forward, i.e. in a self-hosted i.e. dev scenario. But it took the dedi scenario to shake things looks 'permanent'. No idea why. Was experiencing a similar issue with my A3 unit and/or steam group. Again, no idea why. Both appear to land properly in the local squads cached folders afterward. Still the temp folder, pretty sure that is something that gets injected by your environment and/or relayed, IIRC, from other squads HTML file server paths. HTH, 🍻

**Comment by Robnewww** [comment]
Possibly related but I found that if the tag is already taken and thus there is already a *.paa file in the temp folder then it won't display.
